### PR TITLE
[bsp][stm32f10x-HAL][drivers]添加can驱动

### PR DESCRIPTION
**Pull request by ylz0923**
## 拉取/合并请求描述：(PR description)

[
[bsp][stm32f10x-HAL][drivers]添加can驱动, 已在stm32f103rc上测试.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by misonyo** [comment]
@BernardXiong 旧BSP还维护吗，这个PR是向旧的BSP增加驱动

**Comment by BernardXiong** [comment]
走新的BSP即可

File: bsp/stm32/libraries/HAL_Drivers/drv_can.c [hunk 1]
```diff
@@ -0,0 +1,1138 @@
...
+ * Change Logs:
+ * Date           Author       Notes
+ * 2018-08-05     Xeon Xu      the first version
+ * 2019-01-22     YLZ          port from stm324xx-HAL to bsp stm3210x-HAL
+ * 2019-01-26     YLZ          add support EXTID RTR Frame. add filter for canopen.
+ *                             modify send, recv functions and IRQs. fix bug.
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "drv_can.h"
```
Comment (misonyo) [inline bsp/stm32/libraries/HAL_Drivers/drv_can.c]: 此文件请添加RT_USING_CAN条件编译。

File: bsp/stm32/libraries/HAL_Drivers/drv_can.c [hunk 2]
```diff
@@ -0,0 +1,1138 @@
...
+            default_index = index;
+    }
+
+    if(default_index != len)
+        return default_index;
+
+    return 0;
+}
+
+#ifdef USING_BXCAN1
```
Comment (misonyo) [inline bsp/stm32/libraries/HAL_Drivers/drv_can.c]: 宏定义对应修改

File: bsp/stm32/libraries/HAL_Drivers/drv_can.c [hunk 3]
```diff
@@ -0,0 +1,1138 @@
...
+        dev_can1.status.lasterrtype = errtype & 0x70;
+        hcan->Instance->ESR &= ~0x70;
+    }
+    dev_can1.status.rcverrcnt = errtype >> 24;
+    dev_can1.status.snderrcnt = (errtype >> 16 & 0xFF);
+    dev_can1.status.errcode = errtype & 0x07;
+    hcan->Instance->MSR |= CAN_MSR_ERRI;
+    rt_interrupt_leave();
+}
+#endif // USING_BXCAN1
```
Comment (misonyo) [inline bsp/stm32/libraries/HAL_Drivers/drv_can.c]: 宏定义对应修改，CAN2同样需要修改

File: bsp/stm32/libraries/HAL_Drivers/drv_can.c [hunk 4]
```diff
@@ -0,0 +1,1138 @@
...
+            break;
+        case CAN_FIFO1:
+            pRxMsg = hcan->pRx1Msg;
+            break;
+        default:
+            RT_ASSERT(0);
+            break;
+    }
+    
+
```
Comment (misonyo) [inline bsp/stm32/libraries/HAL_Drivers/drv_can.c]: 代码请根据RTT代码规范使用工具格式化，不能有2个相连的空行

File: bsp/stm32/libraries/HAL_Drivers/drv_can.c [hunk 5]
```diff
@@ -0,0 +1,1138 @@
...
+static const struct rt_can_ops drv_can_ops =
+{
+    drv_configure,
+    drv_control,
+    drv_sendmsg,
+    drv_recvmsg,
+};
+
+int hw_can_init(void)
```
Comment (misonyo) [inline bsp/stm32/libraries/HAL_Drivers/drv_can.c]: 这个初始化函数统一按照rt_hw_xxx_init()命名

**Pull request closed without merging**